driver/controller.go

```diff
@@ -179,7 +200,17 @@ func (d *Driver) ControllerPublishVolume(ctx context.Context, req *csi.Controlle
 
 	dropletID, err := strconv.Atoi(req.NodeId)
 	if err != nil {
-		return nil, fmt.Errorf("malformed nodeId %q detected: %s", req.NodeId, err)
+		// don't return because the CSI tests passes ID's in non-integer format.
+		dropletID = 1 // for testing purposes only. Will fail in real world API
+		d.log.WithField("node_id", req.NodeId).Warn("node ID cannot be converted to an integer")
+	}
+
+	if req.Readonly {
+		// TODO(arslan): we should return codes.InvalidArgument, but the CSI
+		// test fails, because according to the CSI Spec, this flag cannot be
+		// changed on the same volume. However we don't use this flag at all,
+		// as there are no `readonly` attachable volumes.
+		return nil, status.Error(codes.AlreadyExists, "read only Volumes are not supported")
```

Comment:
🤔 good point, i wonder if there is a conformance tests for this.. we can check internally while releasing 1.11

Reply:
yeap definitely!